pkg/services/validator/oci.go

```diff
@@ -46,6 +51,50 @@ func readOciPluginRules(vc *components.ValidatorConfig, _ *cfg.TaskConfig, kClie
 	return nil
 }
 
+// configureAuth prompts the user to configure their OCI registry authentication details.
+func configureAuth(r *plug.OciRegistryRule) error {
+	r.Auth.SecretName = nil
+
+	authType, err := prompts.Select("Authentication type", []string{basicAuth, ecrAuth})
+	if err != nil {
+		return err
+	}
+
+	if authType == basicAuth {
+		if r.Auth.Basic == nil {
+			r.Auth.Basic = &plug.BasicAuth{}
+		}
+		r.Auth.Basic.Username, r.Auth.Basic.Password, err = prompts.ReadBasicCreds(
+			"Username", "Password", r.Auth.Basic.Username, r.Auth.Basic.Password, false, false,
+		)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if r.Auth.ECR == nil {
+		r.Auth.ECR = &plug.ECRAuth{}
+	}
+
+	r.Auth.ECR.AccessKeyID, err = prompts.ReadText("Access Key ID", r.Auth.ECR.AccessKeyID, false, -1)
```

Comment:
Do we not already have a function that reads these three pieces of information?

Reply:
I had checked if prompts tui had any aws helpers, but it didn't. We do have `func readAwsCredsHelper(c *components.AWSPluginConfig) error` which is used in the aws plugin. It reads the values right into the AWSPluginConfig. For now i'll make a more generic version of that function, but long term IMO the function should be moved the prompts-tui